privacyidea/lib/eventhandler/webhookevanthandler.py

```diff
@@ -0,0 +1,150 @@
+
+        :return: list of allowed positions
+        """
+        return ["post", "pre"]
+
+    @property
+    def actions(cls):
+        """
+        This method returns a dictionary of allowed actions and possible
+        options in this handler module.
+
+        :return: dict with actions
+        """
+        # The event handler has just one action. Maybe we can  hide action select for the clarity of the UI
+        actions = {ACTION_TYPE.POST_WEBHOOK: {
+            "URL": {
+                "type": "str",
+                "required": True,
+                "description": _("The URL the WebHook is posted to")
+            },
+            "content_type": {
+                "type": "str",
+                "required": True,
+                "description": _("The encoding that is sent to the WebHook, for example json"),
+                "value": [
+                    CONTENT_TYPE.JSON,
+                    CONTENT_TYPE.URLENCODED]
+            },
+            "data": {
+                "type": "str",
+                "required": True,
+                "description": _("The data posted in the WebHook")
+            }
+        }}
+        return actions
+
+    def do(self, action, options=None):
+        """
+        This method executes the defined action in the given event.
+
+        :param action: The action to perform
+        :type action: str
+        :param options: Contains the flask parameters g, URL, data, parameter and the fire and forget
+        :type options: dict
+        :return:
+        """
+        ret = True
+        g = options.get("g")
+        request = options.get("request")
+        handler_def = options.get("handler_def")
+        handler_options = handler_def.get("options")
+        webhook_url = handler_options.get("URL")
+        webhook_text = handler_options.get("data")
+        content_type = handler_options.get("content_type")
+
+        # Replace placeholder in data
+        logged_in_user = g.logged_in_user.get("username")
+        attributes = {
+            'logged_in_user': logged_in_user
+        }
+        webhook_text = webhook_text.format_map(defaultdict(str, attributes))
+
+        if action.lower() == ACTION_TYPE.POST_WEBHOOK:
+            try:
+                if content_type in (CONTENT_TYPE.URLENCODED, CONTENT_TYPE.JSON):
+                    """
+                    This is the main function where the webhook is sent.
+                    Documentation of the requests function is fond her docs.python-requests.org
+                    """
+                    resp = requests.post(webhook_url, data=json.dumps(webhook_text, sort_keys=True, default=str),
+                                         headers={'Content-Type': content_type}, timeout=1.0)
+                    # all answers will be logged in the debug. The HTTP response code will be shown in the audit too
+                    httpcode = resp.status_code
+                    log.info('A webhook is send to {0!r} with the text: {1!r}'.format(
```

Comment:
Please put the log.info in front of the `requests.post`, so that the log file contains the information, that a request is sent - in case the request fails. Thus you have this information in the log file.

Reply:
https://github.com/privacyidea/privacyidea/pull/3154/commits/63296154129247c5d0cf0c73c7119668bd1a02f0